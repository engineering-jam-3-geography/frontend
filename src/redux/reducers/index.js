import {combineReducers} from 'redux';
import modalsReducer from './modals';

const rootReducer = combineReducers({
    modals: modalsReducer,
});

export default rootReducer;
