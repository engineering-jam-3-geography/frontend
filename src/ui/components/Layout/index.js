import React from 'react';
import PropTypes from 'prop-types';
import Header from '../Header';
import Main from '../Main';
import Footer from '../Footer';

import '../../assets/sass/main.scss';
import './index.scss';

/**
 * Renders page Layout
 * @param {Object}
 * @return {JSX}
 */
const Layout = ({
    children
}) => (
    <div className="Layout">
        <Header />

        <Main>
            {children}
        </Main>

        <Footer />
    </div>
);

Layout.propTypes = {
    children: PropTypes.node
};

export default Layout;
