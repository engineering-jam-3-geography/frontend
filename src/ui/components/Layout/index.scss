@import '../../assets/sass/base/variables';
@import '../../assets/sass/base/mixins';

$Layout: '.Layout';

#{$Layout} {
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 100%;
}
