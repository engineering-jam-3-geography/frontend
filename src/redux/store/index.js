import {createStore, applyMiddleware} from 'redux';
import {createBrowserHistory} from 'history';
import {connectRouter, routerMiddleware} from 'connected-react-router';
import thunk from 'redux-thunk';
import {composeWithDevTools} from 'redux-devtools-extension';
import {createLogger} from 'redux-logger';
import rootReducer from '../reducers';
import config from '../../config';

const {basename = ''} = config;

const history = createBrowserHistory({
    basename
});
const router = routerMiddleware(history);
const middleware = [router, thunk];

if (process.env.NODE_ENV === 'development') {
    const logger = createLogger({collapsed: true, diff: true});

    middleware.push(logger);
}

const store = createStore(
    connectRouter(history)(rootReducer),
    composeWithDevTools(
        applyMiddleware(...middleware)
    )
);

export {history};
export default store;
