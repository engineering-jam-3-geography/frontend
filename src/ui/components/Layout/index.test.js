import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import {compose} from 'redux';
import Layout from './';

Enzyme.configure({adapter: new Adapter()});

describe('<Layout /> UI Component', () => {
    const shallowExpect = compose(expect, toJSON, shallow);

    test('Layout.', () => {
        shallowExpect(
            <Layout>
                <div>Here goes children!</div>
            </Layout>
        ).toMatchSnapshot();
    });
});

