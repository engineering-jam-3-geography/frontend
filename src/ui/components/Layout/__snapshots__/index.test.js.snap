// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> UI Component Layout. 1`] = `
<div
  className="Layout"
>
  <Connect(Header) />
  <Main>
    <div>
      Here goes children!
    </div>
  </Main>
  <Footer
    isExtended={false}
  />
  <div
    className="Layout__error"
  >
    <Connect(ErrorMessage) />
  </div>
  <div
    className="Layout__modal"
  >
    <Connect(PasswordModal) />
  </div>
</div>
`;
